### PR TITLE
go/runtime/bundle: Add expected enclave identity to manifest

### LINKED ISSUE #5929: Include expected enclave identity in manifest
Currently the enclave identity is computed on-demand for SGX components. In case of other TEEs like TDX, this is not always possible or may involve a complex process. To this end, we should make expected enclave identity computation a build-time feature and include it in the manifest.

### PR DESCRIPTION
**Pull request by kostko**
Fixes #5929

### CODE REVIEW

File: go/runtime/bundle/bundle.go [hunk 1]
```diff
@@ -219,8 +219,33 @@ func (bnd *Bundle) MrSigner(id component.ID) (*sgx.MrSigner, error) {
...
+
+// EnclaveIdentities returns the enclave identities of the given component.
+func (bnd *Bundle) EnclaveIdentities(id component.ID) ([]sgx.EnclaveIdentity, error) {
+	// If the component has a build-time known expected identity, use it.
+	comp := bnd.Manifest.GetComponentByID(id)
+	if comp == nil {
+		return nil, fmt.Errorf("runtime/bundle: component '%s' not available", id)
+	}
+	if len(comp.Identities) > 0 {
+		ids := make([]sgx.EnclaveIdentity, 0, len(comp.Identities))
```
Comment (peternose) [inline go/runtime/bundle/bundle.go:241]:
```suggestion
		ids := slices.Clone(comp.Identities)
```

Comment (kostko) [reply go/runtime/bundle/bundle.go:241]: Oh, nice!

Comment (kostko) [reply go/runtime/bundle/bundle.go:241]: Oh, but this only works for the same type, these are different now and require a map function as well. Didn't see anything like it in `slices`?

Comment (peternose) [reply go/runtime/bundle/bundle.go:241]: Overlooked `id.Enclave` :facepalm: I think there is no function that behaves like map, so please ignore this comment.

**Pull request merged**